src/lib/uri/uri.c

```diff
@@ -170,6 +179,35 @@ uri_create(struct uri *uri, const char *str)
 	return 0;
 }
 
+static int
+uri_format_param(char **str, int *len, const struct uri_param *param)
+{
+	int total = 0;
+	if (param->value_count == 0)
+		SNPRINT(total, snprintf, *str, *len, "%s", param->name);
+	for (int i = 0; i < param->value_count; i++) {
+		SNPRINT(total, snprintf, *str, *len, "%s=%s",
+			param->name, param->values[i]);
+		if (i != param->value_count - 1)
+			SNPRINT(total, snprintf, *str, *len, "&");
+	}
+	return total;
+}
+
+static int
+uri_format_params(char **str, int *len, const struct uri *uri)
+{
+	int total = 0;
+	if (uri->param_count)
```

Comment:
Please, use explicit `!= 0`. Also you can make `if (uri->param_count == 0) return 0;`. Otherwise the loop below is useless for 0 param count. You do the `if` anyway. Why not return right away?